spi-fly/spi-fly-core/src/main/java/org/apache/aries/spifly/ProviderBundleTrackerCustomizer.java

```diff
@@ -169,6 +171,14 @@ public ProviderBundleTrackerCustomizer(BaseActivator activator, Bundle spiBundle
                         else
                             properties = findServiceRegistrationProperties(bundle.getHeaders(), registrationClassName, className);
 
+                        String[] registrationClassNames;
+                        if (registrationClassName.equals("javax.naming.spi.InitialContextFactory")) {
+                            registrationClassNames = new String[2];
+                            registrationClassNames[1] = cls.getName();
+                        } else {
+                            registrationClassNames = new String[1];
+                        }
+                        registrationClassNames[0] = registrationClassName;

```

Comment:
This change is very specific to the javax.naming.spi.InitialContextFactory and does not belong in SPI-Fly IMHO.